lib/slurm_squeue_client.rb

```diff
@@ -61,7 +61,73 @@ def sinfo
     
     s.success? ? @sinfo = o : raise(CommandFailed, e)
   end
-  
+
+  # Return length of GRES field from SLURM
+  # @return [Integer] gres length
+  def gres_length
+    return @gres_length if definde?(@gres_length)
+
+    Open3.pipeline_rw "sinfo -o '%G' | awk '{ print length }'", "sort -n", "tail -1" do |stdin,stdout|
+      stdin.write stdout
+      stdin.close
+      @gres_length = stdout.read.to_i
+    end
+  end
+
+  # Parse and return total number of GPU nodes in a SLURM cluster
+  # @return [Integer] number of GPU nodes
+  def gpu_nodes
+    return @available_gpu_nodes if defined?(@available_gpu_nodes)
+
+    Open3.pipeline_rw "sinfo -N -h -a --Format='nodehost,gres:#{@gres_length} available'", 'uniq', 'grep gpu:v', 'wc -l' do |stdin, stdout|
```

Comment:
Why does `--Format` include the string `available`?

Remove `v` from `gpu:v` because it's not guaranteed future GPUs will start with v, that's just what's on Pitzer. The GPU nodes on owens will start with `p` as they are `p100`.  So make it just `grep gpu:`.